Replicator/Replicator.cc

```diff
@@ -119,7 +125,7 @@ namespace litecore { namespace repl {
             Assert(_connectionState == Connection::kClosed);
             Signpost::begin(Signpost::replication, uintptr_t(this));
             _connectionState = Connection::kConnecting;
-            connection().start();
+            connection().start(_weakConnectionDelegateThis);
```

Comment:
Maybe setting the delegate in the constructor so the start() doesn't need to change the signature?

Reply:
It is due to a technical reason: WeakHolder is a member of Replicator, and I cannot pass a member in the constructor of the parent class.

Reply:
What about adding a setDelegate method instead, that you can call in the Replicator constructor?

Reply:
I thought about it, too, but, perhaps, randomly, I feel start is also good place to set it: the delegate is not meaningful until the replication is started. It also is consistent with WebSocket::connect(Delegate), i.e., you need a delegate only after connect.